io/teca_cf_reader.cxx

```diff
@@ -1721,6 +1734,16 @@ const_p_teca_dataset teca_cf_reader::execute(unsigned int port,
                 // select the requested time step
                 // subset point centered variables based on the incoming requested
                 // extent.
+
+                 // TODO/FIXME: Currently we assume that the ensemble is the first
```

Comment:
before we close this PR, let's make sure to open an issue to generalize this (this will end up being a useful feature, btw)

Reply:
If I read it correctly, TECA already assumes an implicit order of dimensions. For example it seems to assumes it is always `time`, `z`, `y`, `x` if I read everything correctly. Checking for dimension names is in arbitrary order (unlike what I am doing here). But when assembling `starts` and `counts` arrays, the assumption always seems to be that the dimensions are in that order. I can try to increase the flexibility a bit more by trying to mimic what TECA is doing with other dimensions. Though I do not fully understand all of that. Perhaps we can take a look together on Tuesday when we meet. You may know NetCDF conventions for climate data better than me and may have a better idea why TECA is doing things the way it is.

Reply:
I added a new commit that makes handling this in line with other dimensions. Dimension order now should not matter, but it still assumes a data layout of `ensemble`/`time`/'z'/'y'/'x'. Since TECA already did assume a layout of `time`/'z'/'y'/'x' this seems reasonable to me. Let me know if we should still do an issue.